mws/resource_mws_permission_assignment.go

```diff
@@ -0,0 +1,143 @@
+package mws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func NewPermissionAssignmentAPI(ctx context.Context, m any) PermissionAssignmentAPI {
+	return PermissionAssignmentAPI{m.(*common.DatabricksClient), ctx}
+}
+
+// NetworksAPI exposes the mws networks API
+type PermissionAssignmentAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+type Permissions struct {
+	Permissions []string `json:"permissions"`
+}
+
+func (a PermissionAssignmentAPI) CreateOrUpdate(workspaceId, principalId int64, r Permissions) error {
+	if a.client.AccountID == "" {
+		return errors.New("must have `account_id` on provider")
+	}
+	path := fmt.Sprintf(
+		"/preview/accounts/%s/workspaces/%d/permissionassignments/principals/%d",
+		a.client.AccountID, workspaceId, principalId)
+	return a.client.Put(a.context, path, r)
+}
+
+func (a PermissionAssignmentAPI) Remove(workspaceId, principalId string) error {
+	if a.client.AccountID == "" {
+		return errors.New("must have `account_id` on provider")
+	}
+	path := fmt.Sprintf(
+		"/preview/accounts/%s/workspaces/%s/permissionassignments/principals/%s",
+		a.client.AccountID, workspaceId, principalId)
+	return a.client.Delete(a.context, path, nil)
+}
+
+type Principal struct {
```

Comment:
is there a way to share this struct with access/resource_permissionassignment.go?

Reply:
no, because generally these two packages are unrelated. this will change with SDK